utilities/oadp.py

```
@@ -77,11 +120,12 @@ def __enter__(self) -> "VeleroBackup":
     def __exit__(self, exception_type, exception_value, traceback) -> None:
         try:
             if self.teardown:
-                delete_velero_resource(resource=self, client=self.client)
+                try:
+                    delete_velero_resource(resource=self, client=self.client)
+                except Exception as e:
+                    raise ResourceTeardownError(resource=self) from e
             else:
                 LOGGER.info(f"Skipping Velero delete for {self.kind} {self.name} (teardown=False)")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)